src/js/base/runtime.js

```diff
@@ -5831,6 +5872,19 @@ function (Namespace, jsnums, codePoint, seedrandom, util) {
         thisRuntime[nameMap[longName]] = thisRuntime[longName];
     }
 
+    thisRuntime.bounceAllowed = true; // Default
+    if (theOutsideWorld.options) {
+      thisRuntime.bounceAllowed = theOutsideWorld.options.bounceAllowed;
+    }
+
+    // In principle it should be possible for a parent runtime to have bouncing off
+    // but a "child runtime" have bouncing on, though it might require some nasty
+    // modifications to the run function. For now we just require that the child runtime
+    // can only allow bounces if the parent runtime can as well
```

Comment:
Oh wow, you're definitely right. I don't see any way we could do bounces from a "child" runtime which has a non-bouncing parent. I added a comment explaining this.